toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -57,10 +60,179 @@ func handleKernelCommandLine(extraCommandLine string, imageChroot *safechroot.Ch
+	logger.Log.Debugf("Get existing SELinux mode")
+
+	grub2Config, err := readGrub2ConfigFile(imageChroot)
+	if err != nil {
+		return "", err
+	}
+
+	// Try to find any existing SELinux args.
+	selinuxArgsMatch := linuxCommandLineSELinuxRegex.FindStringSubmatchIndex(grub2Config)
+	if selinuxArgsMatch == nil {
+		// SELinux not present in the kernel command-line.
+		// Therefore, it is disabled.
+		return imagecustomizerapi.SELinuxDisabled, nil
+	}
+
+	// Check if "enforcing=1" is present.
+	// Note: A negative index means the sub-match is not present.
+	// Note: regexp returns index pairs. So, [4] is start index of the 2nd sub-match.
+	if selinuxArgsMatch[4] >= 0 {
+		return imagecustomizerapi.SELinuxForceEnforcing, nil
+	}
+
+	selinuxMode, err := getSELinuxModeFromConfigFile(imageChroot)
+	if err != nil {
+		return "", err
+	}
+
+	return selinuxMode, nil
+}
+
+func getSELinuxModeFromConfigFile(imageChroot *safechroot.Chroot) (imagecustomizerapi.SELinux, error) {
+	selinuxConfigFilePath := filepath.Join(imageChroot.RootDir(), installutils.SELinuxConfigFile)
+
+	// Read the SELinux config file.
+	fileBytes, err := os.ReadFile(selinuxConfigFilePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read existing grub2 config file:\n%w", err)
+	}
+
+	selinuxConfig := string(fileBytes)
+
+	match := selinuxConfigModeRegex.FindStringSubmatch(selinuxConfig)
+	if match == nil {
+		return "", fmt.Errorf("failed to find SELinux mode in %s file", installutils.SELinuxConfigFile)
+	}
+
+	selinuxConfigMode := match[1]
+
+	switch selinuxConfigMode {
+	case "enforcing":
+		return imagecustomizerapi.SELinuxEnforcing, nil
+
+	case "permissive":
+		return imagecustomizerapi.SELinuxPermissive, nil
+
+	case "disabled":
+		return imagecustomizerapi.SELinuxDisabled, nil
+
+	default:
+		return "", fmt.Errorf("unknown SELinux mode (%s) found in (%s) file", selinuxConfigMode,
+			installutils.SELinuxConfigFile)
+	}
+}
+
+func readGrub2ConfigFile(imageChroot *safechroot.Chroot) (string, error) {
+	logger.Log.Debugf("Reading grub.cfg file")
+
+	grub2ConfigFilePath := getGrub2ConfigFilePath(imageChroot)
+
+	// Read the existing grub.cfg file.
+	grub2ConfigFileBytes, err := os.ReadFile(grub2ConfigFilePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read existing grub2 config file:\n%w", err)
+	}
+
+	grub2Config := string(grub2ConfigFileBytes)
+
+	return grub2Config, nil
+}
+
+func writeGrub2ConfigFile(grub2Config string, imageChroot *safechroot.Chroot) error {
+	logger.Log.Debugf("Writing grub.cfg file")
+
+	grub2ConfigFilePath := getGrub2ConfigFilePath(imageChroot)
+
+	// Update grub.cfg file.
+	err := os.WriteFile(grub2ConfigFilePath, []byte(grub2Config), 0)
 	if err != nil {
-		return fmt.Errorf("failed to write new grub2 config file: %w", err)
+		return fmt.Errorf("failed to write new grub2 config file:\n%w", err)
 	}
 
 	return nil
 }
+
+func getGrub2ConfigFilePath(imageChroot *safechroot.Chroot) string {
```

Comment:
these grub helpers can be useful to others who deal with grub. Does it make sense to  move them into their own file/object? (I have similar needs with the iso generation code)

Reply:
All the code in `customizeboot.go` is related to the grub.cfg file. So, it is already separated out.

If you ever need these functions to live in a shared module, then we can cross that bridge when we come to it.